File: src/main/java/mnm/mods/tabbychat/api/Message.java
package mnm.mods.tabbychat.api;

import java.time.LocalDateTime;

import net.minecraft.util.text.ITextComponent;

/**
 * Represents a message.
 */
public interface Message {

    /**
     * Gets the message
     *
     * @return The message
     */
    ITextComponent getMessage();

    /**
     * Gets the date that this message was sent.
     *
     * @return The date
     */
    LocalDateTime getDateTime();

}

File: src/main/kotlin/mnm/mods/tabbychat/api/Message.kt
package mnm.mods.tabbychat.api

import net.minecraft.util.text.ITextComponent
import java.time.LocalDateTime

/**
 * Represents a message.
 */
interface Message {

    /**
     * Gets the message
     *
     * @return The message
     */
    val message: ITextComponent

    /**
     * Gets the date that this message was sent.
     *
     * @return The date
     */
    val dateTime: LocalDateTime?

}
